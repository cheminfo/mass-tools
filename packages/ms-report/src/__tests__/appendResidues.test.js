const appendResidues = require('../appendResidues');

describe('appendResidues', () => {
  it('AA(Ph)A(Ph)A(Ts)A(Ph)U', () => {
    let data = {};
    appendResidues(data, 'AA(Ph)A(Ph)A(Ts)A(Ph)U');
    expect(data).toMatchSnapshot();
  });

  it('HAlaAlaAlaOH', () => {
    let data = {};
    appendResidues(data, 'HAlaAlaAlaOH');
    expect(data).toMatchSnapshot();
  });

  it('HAlaAla(H-1OH)AlaOH', () => {
    let data = {};
    appendResidues(data, 'HAlaAla(H-1OH)AlaOH');
    expect(data).toMatchSnapshot();
  });

  it('H(+)AlaAla(H-1OH)AlaOH', () => {
    let data = {};
    appendResidues(data, 'H(+)AlaAla(H-1OH)AlaOH');
    expect(data).toMatchSnapshot();
  });

  it('ForAlaAla(H-1OH)AlaOH', () => {
    let data = {};
    appendResidues(data, '(For)AlaAla(H-1OH)AlaOH');
    expect(data).toMatchSnapshot();
  });

  it('(MeO)AA(H-1NH2)AA rna', () => {
    let data = {};
    appendResidues(data, '(MeO)AA(H-1NH2)AA', { kind: 'rna' });
    expect(data).toMatchSnapshot();
  });

  it('(C5H5O)IleLeuAspAspLeuCys(C26H30N2OSi)AlaAsnGlnLeuGlnProLeuLeuLeuLysOH', () => {
    let data = {};
    appendResidues(
      data,
      '(C5H5O)IleLeuAspAspLeuCys(C26H30N2OSi)AlaAsnGlnLeuGlnProLeuLeuLeuLysOH',
      { kind: 'peptide' },
    );
    expect(data).toMatchSnapshot();
  });
});
