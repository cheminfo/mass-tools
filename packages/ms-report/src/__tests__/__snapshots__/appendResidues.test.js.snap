// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`appendResidues (C5H5O)IleLeuAspAspLeuCys(C26H30N2OSi)AlaAsnGlnLeuGlnProLeuLeuLeuLysOH 1`] = `
Object {
  "residues": Object {
    "all": Array [
      Object {
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "begin",
        "label": "Nter",
      },
      Object {
        "fromBegin": 1,
        "fromEnd": 16,
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "residue",
        "label": "I",
        "results": Object {
          "begin": Array [],
          "end": Array [],
        },
        "value": "Ile",
      },
      Object {
        "fromBegin": 2,
        "fromEnd": 15,
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "residue",
        "label": "L",
        "results": Object {
          "begin": Array [],
          "end": Array [],
        },
        "value": "Leu",
      },
      Object {
        "fromBegin": 3,
        "fromEnd": 14,
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "residue",
        "label": "D",
        "results": Object {
          "begin": Array [],
          "end": Array [],
        },
        "value": "Asp",
      },
      Object {
        "fromBegin": 4,
        "fromEnd": 13,
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "residue",
        "label": "D",
        "results": Object {
          "begin": Array [],
          "end": Array [],
        },
        "value": "Asp",
      },
      Object {
        "fromBegin": 5,
        "fromEnd": 12,
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "residue",
        "label": "L",
        "results": Object {
          "begin": Array [],
          "end": Array [],
        },
        "value": "Leu",
      },
      Object {
        "fromBegin": 6,
        "fromEnd": 11,
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "residue",
        "label": "ς",
        "replaced": true,
        "results": Object {
          "begin": Array [],
          "end": Array [],
        },
        "value": "Cys(C26H30N2OSi)",
      },
      Object {
        "fromBegin": 7,
        "fromEnd": 10,
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "residue",
        "label": "A",
        "results": Object {
          "begin": Array [],
          "end": Array [],
        },
        "value": "Ala",
      },
      Object {
        "fromBegin": 8,
        "fromEnd": 9,
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "residue",
        "label": "N",
        "results": Object {
          "begin": Array [],
          "end": Array [],
        },
        "value": "Asn",
      },
      Object {
        "fromBegin": 9,
        "fromEnd": 8,
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "residue",
        "label": "Q",
        "results": Object {
          "begin": Array [],
          "end": Array [],
        },
        "value": "Gln",
      },
      Object {
        "fromBegin": 10,
        "fromEnd": 7,
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "residue",
        "label": "L",
        "results": Object {
          "begin": Array [],
          "end": Array [],
        },
        "value": "Leu",
      },
      Object {
        "fromBegin": 11,
        "fromEnd": 6,
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "residue",
        "label": "Q",
        "results": Object {
          "begin": Array [],
          "end": Array [],
        },
        "value": "Gln",
      },
      Object {
        "fromBegin": 12,
        "fromEnd": 5,
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "residue",
        "label": "P",
        "results": Object {
          "begin": Array [],
          "end": Array [],
        },
        "value": "Pro",
      },
      Object {
        "fromBegin": 13,
        "fromEnd": 4,
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "residue",
        "label": "L",
        "results": Object {
          "begin": Array [],
          "end": Array [],
        },
        "value": "Leu",
      },
      Object {
        "fromBegin": 14,
        "fromEnd": 3,
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "residue",
        "label": "L",
        "results": Object {
          "begin": Array [],
          "end": Array [],
        },
        "value": "Leu",
      },
      Object {
        "fromBegin": 15,
        "fromEnd": 2,
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "residue",
        "label": "L",
        "results": Object {
          "begin": Array [],
          "end": Array [],
        },
        "value": "Leu",
      },
      Object {
        "fromBegin": 16,
        "fromEnd": 1,
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "residue",
        "label": "K",
        "results": Object {
          "begin": Array [],
          "end": Array [],
        },
        "value": "Lys",
      },
      Object {
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "end",
        "label": "OH",
      },
    ],
    "alternatives": Object {
      "ς": Object {
        "count": 1,
      },
    },
    "begin": Object {
      "info": Object {
        "nbOver": 0,
        "nbUnder": 0,
      },
      "kind": "begin",
      "label": "Nter",
    },
    "end": Object {
      "info": Object {
        "nbOver": 0,
        "nbUnder": 0,
      },
      "kind": "end",
      "label": "OH",
    },
    "replacements": Object {
      "C5H5O": Object {
        "label": "Nter",
      },
      "Cys(C26H30N2OSi)": Object {
        "label": "ς",
        "modification": "C26H30N2OSi",
        "residue": "Cys",
      },
    },
    "residues": Array [
      Object {
        "fromBegin": 1,
        "fromEnd": 16,
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "residue",
        "label": "I",
        "results": Object {
          "begin": Array [],
          "end": Array [],
        },
        "value": "Ile",
      },
      Object {
        "fromBegin": 2,
        "fromEnd": 15,
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "residue",
        "label": "L",
        "results": Object {
          "begin": Array [],
          "end": Array [],
        },
        "value": "Leu",
      },
      Object {
        "fromBegin": 3,
        "fromEnd": 14,
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "residue",
        "label": "D",
        "results": Object {
          "begin": Array [],
          "end": Array [],
        },
        "value": "Asp",
      },
      Object {
        "fromBegin": 4,
        "fromEnd": 13,
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "residue",
        "label": "D",
        "results": Object {
          "begin": Array [],
          "end": Array [],
        },
        "value": "Asp",
      },
      Object {
        "fromBegin": 5,
        "fromEnd": 12,
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "residue",
        "label": "L",
        "results": Object {
          "begin": Array [],
          "end": Array [],
        },
        "value": "Leu",
      },
      Object {
        "fromBegin": 6,
        "fromEnd": 11,
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "residue",
        "label": "ς",
        "replaced": true,
        "results": Object {
          "begin": Array [],
          "end": Array [],
        },
        "value": "Cys(C26H30N2OSi)",
      },
      Object {
        "fromBegin": 7,
        "fromEnd": 10,
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "residue",
        "label": "A",
        "results": Object {
          "begin": Array [],
          "end": Array [],
        },
        "value": "Ala",
      },
      Object {
        "fromBegin": 8,
        "fromEnd": 9,
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "residue",
        "label": "N",
        "results": Object {
          "begin": Array [],
          "end": Array [],
        },
        "value": "Asn",
      },
      Object {
        "fromBegin": 9,
        "fromEnd": 8,
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "residue",
        "label": "Q",
        "results": Object {
          "begin": Array [],
          "end": Array [],
        },
        "value": "Gln",
      },
      Object {
        "fromBegin": 10,
        "fromEnd": 7,
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "residue",
        "label": "L",
        "results": Object {
          "begin": Array [],
          "end": Array [],
        },
        "value": "Leu",
      },
      Object {
        "fromBegin": 11,
        "fromEnd": 6,
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "residue",
        "label": "Q",
        "results": Object {
          "begin": Array [],
          "end": Array [],
        },
        "value": "Gln",
      },
      Object {
        "fromBegin": 12,
        "fromEnd": 5,
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "residue",
        "label": "P",
        "results": Object {
          "begin": Array [],
          "end": Array [],
        },
        "value": "Pro",
      },
      Object {
        "fromBegin": 13,
        "fromEnd": 4,
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "residue",
        "label": "L",
        "results": Object {
          "begin": Array [],
          "end": Array [],
        },
        "value": "Leu",
      },
      Object {
        "fromBegin": 14,
        "fromEnd": 3,
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "residue",
        "label": "L",
        "results": Object {
          "begin": Array [],
          "end": Array [],
        },
        "value": "Leu",
      },
      Object {
        "fromBegin": 15,
        "fromEnd": 2,
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "residue",
        "label": "L",
        "results": Object {
          "begin": Array [],
          "end": Array [],
        },
        "value": "Leu",
      },
      Object {
        "fromBegin": 16,
        "fromEnd": 1,
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "residue",
        "label": "K",
        "results": Object {
          "begin": Array [],
          "end": Array [],
        },
        "value": "Lys",
      },
    ],
  },
}
`;

exports[`appendResidues (MeO)AA(H-1NH2)AA rna 1`] = `
Object {
  "residues": Object {
    "all": Array [
      Object {
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "begin",
        "label": "5'",
      },
      Object {
        "fromBegin": 1,
        "fromEnd": 4,
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "residue",
        "label": "A",
        "results": Object {
          "begin": Array [],
          "end": Array [],
        },
        "value": "Amp",
      },
      Object {
        "fromBegin": 2,
        "fromEnd": 3,
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "residue",
        "label": "α",
        "replaced": true,
        "results": Object {
          "begin": Array [],
          "end": Array [],
        },
        "value": "Amp(H-1NH2)",
      },
      Object {
        "fromBegin": 3,
        "fromEnd": 2,
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "residue",
        "label": "A",
        "results": Object {
          "begin": Array [],
          "end": Array [],
        },
        "value": "Amp",
      },
      Object {
        "fromBegin": 4,
        "fromEnd": 1,
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "residue",
        "label": "A",
        "results": Object {
          "begin": Array [],
          "end": Array [],
        },
        "value": "Amp",
      },
      Object {
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "end",
        "label": "H",
      },
    ],
    "alternatives": Object {
      "α": Object {
        "count": 1,
      },
    },
    "begin": Object {
      "info": Object {
        "nbOver": 0,
        "nbUnder": 0,
      },
      "kind": "begin",
      "label": "5'",
    },
    "end": Object {
      "info": Object {
        "nbOver": 0,
        "nbUnder": 0,
      },
      "kind": "end",
      "label": "H",
    },
    "replacements": Object {
      "Amp(H-1NH2)": Object {
        "label": "α",
        "modification": "H-1NH2",
        "residue": "Amp",
      },
      "MeO": Object {
        "label": "5'",
      },
    },
    "residues": Array [
      Object {
        "fromBegin": 1,
        "fromEnd": 4,
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "residue",
        "label": "A",
        "results": Object {
          "begin": Array [],
          "end": Array [],
        },
        "value": "Amp",
      },
      Object {
        "fromBegin": 2,
        "fromEnd": 3,
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "residue",
        "label": "α",
        "replaced": true,
        "results": Object {
          "begin": Array [],
          "end": Array [],
        },
        "value": "Amp(H-1NH2)",
      },
      Object {
        "fromBegin": 3,
        "fromEnd": 2,
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "residue",
        "label": "A",
        "results": Object {
          "begin": Array [],
          "end": Array [],
        },
        "value": "Amp",
      },
      Object {
        "fromBegin": 4,
        "fromEnd": 1,
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "residue",
        "label": "A",
        "results": Object {
          "begin": Array [],
          "end": Array [],
        },
        "value": "Amp",
      },
    ],
  },
}
`;

exports[`appendResidues AA(Ph)A(Ph)A(Ts)A(Ph)U 1`] = `
Object {
  "residues": Object {
    "all": Array [
      Object {
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "begin",
        "label": "H",
      },
      Object {
        "fromBegin": 1,
        "fromEnd": 6,
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "residue",
        "label": "A",
        "results": Object {
          "begin": Array [],
          "end": Array [],
        },
        "value": "Ala",
      },
      Object {
        "fromBegin": 2,
        "fromEnd": 5,
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "residue",
        "label": "α",
        "replaced": true,
        "results": Object {
          "begin": Array [],
          "end": Array [],
        },
        "value": "Ala(Ph)",
      },
      Object {
        "fromBegin": 3,
        "fromEnd": 4,
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "residue",
        "label": "α",
        "replaced": true,
        "results": Object {
          "begin": Array [],
          "end": Array [],
        },
        "value": "Ala(Ph)",
      },
      Object {
        "fromBegin": 4,
        "fromEnd": 3,
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "residue",
        "label": "¹α",
        "replaced": true,
        "results": Object {
          "begin": Array [],
          "end": Array [],
        },
        "value": "Ala(Ts)",
      },
      Object {
        "fromBegin": 5,
        "fromEnd": 2,
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "residue",
        "label": "α",
        "replaced": true,
        "results": Object {
          "begin": Array [],
          "end": Array [],
        },
        "value": "Ala(Ph)",
      },
      Object {
        "fromBegin": 6,
        "fromEnd": 1,
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "residue",
        "label": "Θ",
        "replaced": true,
        "results": Object {
          "begin": Array [],
          "end": Array [],
        },
        "value": "Sec",
      },
      Object {
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "end",
        "label": "OH",
      },
    ],
    "alternatives": Object {
      "α": Object {
        "count": 2,
      },
    },
    "begin": Object {
      "info": Object {
        "nbOver": 0,
        "nbUnder": 0,
      },
      "kind": "begin",
      "label": "H",
    },
    "end": Object {
      "info": Object {
        "nbOver": 0,
        "nbUnder": 0,
      },
      "kind": "end",
      "label": "OH",
    },
    "replacements": Object {
      "Ala(Ph)": Object {
        "label": "α",
        "modification": "Ph",
        "residue": "Ala",
      },
      "Ala(Ts)": Object {
        "label": "¹α",
        "modification": "Ts",
        "residue": "Ala",
      },
      "Sec": Object {
        "id": "Sec",
        "label": "Θ",
      },
    },
    "residues": Array [
      Object {
        "fromBegin": 1,
        "fromEnd": 6,
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "residue",
        "label": "A",
        "results": Object {
          "begin": Array [],
          "end": Array [],
        },
        "value": "Ala",
      },
      Object {
        "fromBegin": 2,
        "fromEnd": 5,
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "residue",
        "label": "α",
        "replaced": true,
        "results": Object {
          "begin": Array [],
          "end": Array [],
        },
        "value": "Ala(Ph)",
      },
      Object {
        "fromBegin": 3,
        "fromEnd": 4,
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "residue",
        "label": "α",
        "replaced": true,
        "results": Object {
          "begin": Array [],
          "end": Array [],
        },
        "value": "Ala(Ph)",
      },
      Object {
        "fromBegin": 4,
        "fromEnd": 3,
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "residue",
        "label": "¹α",
        "replaced": true,
        "results": Object {
          "begin": Array [],
          "end": Array [],
        },
        "value": "Ala(Ts)",
      },
      Object {
        "fromBegin": 5,
        "fromEnd": 2,
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "residue",
        "label": "α",
        "replaced": true,
        "results": Object {
          "begin": Array [],
          "end": Array [],
        },
        "value": "Ala(Ph)",
      },
      Object {
        "fromBegin": 6,
        "fromEnd": 1,
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "residue",
        "label": "Θ",
        "replaced": true,
        "results": Object {
          "begin": Array [],
          "end": Array [],
        },
        "value": "Sec",
      },
    ],
  },
}
`;

exports[`appendResidues ForAlaAla(H-1OH)AlaOH 1`] = `
Object {
  "residues": Object {
    "all": Array [
      Object {
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "begin",
        "label": "5'",
      },
      Object {
        "fromBegin": 1,
        "fromEnd": 3,
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "residue",
        "label": "A",
        "results": Object {
          "begin": Array [],
          "end": Array [],
        },
        "value": "Ala",
      },
      Object {
        "fromBegin": 2,
        "fromEnd": 2,
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "residue",
        "label": "α",
        "replaced": true,
        "results": Object {
          "begin": Array [],
          "end": Array [],
        },
        "value": "Ala(H-1OH)",
      },
      Object {
        "fromBegin": 3,
        "fromEnd": 1,
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "residue",
        "label": "A",
        "results": Object {
          "begin": Array [],
          "end": Array [],
        },
        "value": "Ala",
      },
      Object {
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "end",
        "label": "OH",
      },
    ],
    "alternatives": Object {
      "α": Object {
        "count": 1,
      },
    },
    "begin": Object {
      "info": Object {
        "nbOver": 0,
        "nbUnder": 0,
      },
      "kind": "begin",
      "label": "5'",
    },
    "end": Object {
      "info": Object {
        "nbOver": 0,
        "nbUnder": 0,
      },
      "kind": "end",
      "label": "OH",
    },
    "replacements": Object {
      "Ala(H-1OH)": Object {
        "label": "α",
        "modification": "H-1OH",
        "residue": "Ala",
      },
      "For": Object {
        "label": "5'",
      },
    },
    "residues": Array [
      Object {
        "fromBegin": 1,
        "fromEnd": 3,
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "residue",
        "label": "A",
        "results": Object {
          "begin": Array [],
          "end": Array [],
        },
        "value": "Ala",
      },
      Object {
        "fromBegin": 2,
        "fromEnd": 2,
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "residue",
        "label": "α",
        "replaced": true,
        "results": Object {
          "begin": Array [],
          "end": Array [],
        },
        "value": "Ala(H-1OH)",
      },
      Object {
        "fromBegin": 3,
        "fromEnd": 1,
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "residue",
        "label": "A",
        "results": Object {
          "begin": Array [],
          "end": Array [],
        },
        "value": "Ala",
      },
    ],
  },
}
`;

exports[`appendResidues H(+)AlaAla(H-1OH)AlaOH 1`] = `
Object {
  "residues": Object {
    "all": Array [
      Object {
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "begin",
        "label": "5'",
      },
      Object {
        "fromBegin": 1,
        "fromEnd": 3,
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "residue",
        "label": "A",
        "results": Object {
          "begin": Array [],
          "end": Array [],
        },
        "value": "Ala",
      },
      Object {
        "fromBegin": 2,
        "fromEnd": 2,
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "residue",
        "label": "α",
        "replaced": true,
        "results": Object {
          "begin": Array [],
          "end": Array [],
        },
        "value": "Ala(H-1OH)",
      },
      Object {
        "fromBegin": 3,
        "fromEnd": 1,
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "residue",
        "label": "A",
        "results": Object {
          "begin": Array [],
          "end": Array [],
        },
        "value": "Ala",
      },
      Object {
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "end",
        "label": "OH",
      },
    ],
    "alternatives": Object {
      "α": Object {
        "count": 1,
      },
    },
    "begin": Object {
      "info": Object {
        "nbOver": 0,
        "nbUnder": 0,
      },
      "kind": "begin",
      "label": "5'",
    },
    "end": Object {
      "info": Object {
        "nbOver": 0,
        "nbUnder": 0,
      },
      "kind": "end",
      "label": "OH",
    },
    "replacements": Object {
      "Ala(H-1OH)": Object {
        "label": "α",
        "modification": "H-1OH",
        "residue": "Ala",
      },
      "H(+)": Object {
        "label": "5'",
      },
    },
    "residues": Array [
      Object {
        "fromBegin": 1,
        "fromEnd": 3,
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "residue",
        "label": "A",
        "results": Object {
          "begin": Array [],
          "end": Array [],
        },
        "value": "Ala",
      },
      Object {
        "fromBegin": 2,
        "fromEnd": 2,
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "residue",
        "label": "α",
        "replaced": true,
        "results": Object {
          "begin": Array [],
          "end": Array [],
        },
        "value": "Ala(H-1OH)",
      },
      Object {
        "fromBegin": 3,
        "fromEnd": 1,
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "residue",
        "label": "A",
        "results": Object {
          "begin": Array [],
          "end": Array [],
        },
        "value": "Ala",
      },
    ],
  },
}
`;

exports[`appendResidues HAlaAla(H-1OH)AlaOH 1`] = `
Object {
  "residues": Object {
    "all": Array [
      Object {
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "begin",
        "label": "H",
      },
      Object {
        "fromBegin": 1,
        "fromEnd": 3,
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "residue",
        "label": "A",
        "results": Object {
          "begin": Array [],
          "end": Array [],
        },
        "value": "Ala",
      },
      Object {
        "fromBegin": 2,
        "fromEnd": 2,
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "residue",
        "label": "α",
        "replaced": true,
        "results": Object {
          "begin": Array [],
          "end": Array [],
        },
        "value": "Ala(H-1OH)",
      },
      Object {
        "fromBegin": 3,
        "fromEnd": 1,
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "residue",
        "label": "A",
        "results": Object {
          "begin": Array [],
          "end": Array [],
        },
        "value": "Ala",
      },
      Object {
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "end",
        "label": "OH",
      },
    ],
    "alternatives": Object {
      "α": Object {
        "count": 1,
      },
    },
    "begin": Object {
      "info": Object {
        "nbOver": 0,
        "nbUnder": 0,
      },
      "kind": "begin",
      "label": "H",
    },
    "end": Object {
      "info": Object {
        "nbOver": 0,
        "nbUnder": 0,
      },
      "kind": "end",
      "label": "OH",
    },
    "replacements": Object {
      "Ala(H-1OH)": Object {
        "label": "α",
        "modification": "H-1OH",
        "residue": "Ala",
      },
    },
    "residues": Array [
      Object {
        "fromBegin": 1,
        "fromEnd": 3,
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "residue",
        "label": "A",
        "results": Object {
          "begin": Array [],
          "end": Array [],
        },
        "value": "Ala",
      },
      Object {
        "fromBegin": 2,
        "fromEnd": 2,
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "residue",
        "label": "α",
        "replaced": true,
        "results": Object {
          "begin": Array [],
          "end": Array [],
        },
        "value": "Ala(H-1OH)",
      },
      Object {
        "fromBegin": 3,
        "fromEnd": 1,
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "residue",
        "label": "A",
        "results": Object {
          "begin": Array [],
          "end": Array [],
        },
        "value": "Ala",
      },
    ],
  },
}
`;

exports[`appendResidues HAlaAlaAlaOH 1`] = `
Object {
  "residues": Object {
    "all": Array [
      Object {
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "begin",
        "label": "H",
      },
      Object {
        "fromBegin": 1,
        "fromEnd": 3,
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "residue",
        "label": "A",
        "results": Object {
          "begin": Array [],
          "end": Array [],
        },
        "value": "Ala",
      },
      Object {
        "fromBegin": 2,
        "fromEnd": 2,
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "residue",
        "label": "A",
        "results": Object {
          "begin": Array [],
          "end": Array [],
        },
        "value": "Ala",
      },
      Object {
        "fromBegin": 3,
        "fromEnd": 1,
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "residue",
        "label": "A",
        "results": Object {
          "begin": Array [],
          "end": Array [],
        },
        "value": "Ala",
      },
      Object {
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "end",
        "label": "OH",
      },
    ],
    "alternatives": Object {},
    "begin": Object {
      "info": Object {
        "nbOver": 0,
        "nbUnder": 0,
      },
      "kind": "begin",
      "label": "H",
    },
    "end": Object {
      "info": Object {
        "nbOver": 0,
        "nbUnder": 0,
      },
      "kind": "end",
      "label": "OH",
    },
    "replacements": Object {},
    "residues": Array [
      Object {
        "fromBegin": 1,
        "fromEnd": 3,
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "residue",
        "label": "A",
        "results": Object {
          "begin": Array [],
          "end": Array [],
        },
        "value": "Ala",
      },
      Object {
        "fromBegin": 2,
        "fromEnd": 2,
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "residue",
        "label": "A",
        "results": Object {
          "begin": Array [],
          "end": Array [],
        },
        "value": "Ala",
      },
      Object {
        "fromBegin": 3,
        "fromEnd": 1,
        "info": Object {
          "nbOver": 0,
          "nbUnder": 0,
        },
        "kind": "residue",
        "label": "A",
        "results": Object {
          "begin": Array [],
          "end": Array [],
        },
        "value": "Ala",
      },
    ],
  },
}
`;
